java-client/src/main/java/org/opensearch/client/transport/httpclient5/ApacheHttpClient5Options.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.transport.httpclient5;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.hc.client5.http.config.RequestConfig;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.message.BasicHeader;
+import org.apache.hc.core5.http.nio.AsyncResponseConsumer;
+import org.opensearch.client.transport.TransportOptions;
+import org.opensearch.client.transport.Version;
+
+public class ApacheHttpClient5Options implements TransportOptions {
+    private static final String USER_AGENT = "User-Agent";
```

Comment:
I found it interesting that the client [does not declare things like user-agent as constants](https://github.com/search?q=repo%3Aapache%2Fhttpcomponents-client+%22User-Agent%22+language%3AJava&type=code&l=Java). Since this is only used once, maybe we should too? I noticed you don't for "Accept" :)

Reply:
I have largely copied it from `RestClientOptions`, we probably should have constants, I agree

Reply:
Done